src/com/mostc/pftt/util/HostEnvUtil.java

```diff
@@ -229,10 +226,15 @@ private static void installAndConfigureMySql(FileSystemScenario fs, AHost host,
 			String data_dir = Dir_Mysql_5_7 + "\\data";
 			createDirectoryIfNotExists(fs, cm, data_dir);
 			
+			// Allow MySql installer through the Firewall
+			cm.println(EPrintType.IN_PROGRESS,  HostEnvUtil.class,  "Allowing MySql Server through Windows Firewall");
+			host.execElevated(cm, HostEnvUtil.class, "netsh advfirewall firewall add rule name=mysqld dir=in action=allow "
+					+ "program=\""+ Exe_Mysql_5_7_mysqld +"\" enable=yes", AHost.ONE_MINUTE);
+
```

Comment:
Would it make sense to centralize the handling of this part? What happens, if `netsh advfirewall ...` is called with the same parameters multiple times? Wouldn't it add same rule over again, or fail, etc? Maybe it would make sense to encapsulate this in some dedicated method/class, as more is likely to come for IIS and other apps.

Thanks.

Reply:
If it was called multiple times it would have added multiple rules with the same name. They don't get overridden. If VC15 x64 (or any other one) was already installed, it shouldn't be trying to add multiple rules with the same name. However, to be safe and prevent this from happening, I check if the rule exists then do the action depending on it. 

I also moved the code to a dedicated method.

Reply:
Why we should care about it, is because there will be cases where it's not necessarily bound to the binary. Or, some program might need to be upgraded, in which case it might need to be uninstalled/removed first. In that case, setup might do exactly that - adding dup firewall rules or redoing any other steps that might be not needed.

Thanks.